scripts/question_answering/data_pipeline.py

```diff
@@ -0,0 +1,900 @@
+        Returns
+        -------
+        train_examples : List[Tuple]
+            Processed train examples. Each tuple consists of question_id, record_index,
+            context_tokens_indices, question_tokens_indices, context_chars_indices,
+            question_char_indices, start_token_index_of_the_answer, end_token_index_of_the_answer,
+            context, context_tokens_spans
+        dev_examples : List[Tuple]
+            Processed dev examples. Each tuple consists of question_id, record_index,
+            context_tokens_indices, question_tokens_indices, context_chars_indices,
+            question_char_indices, start_token_index_of_the_answer, end_token_index_of_the_answer,
+            context, context_tokens_spans
+        word_vocab : Vocab
+            Word-level vocabulary
+        char_vocab : Vocab
+            Char-level vocabulary
+        """
+        with open(os.path.join(self._data_root_path, self._processed_train_data_file_name),
+                  'r') as f:
+            train_examples = json.load(f)
+
+        with open(os.path.join(self._data_root_path, self._processed_dev_data_file_name), 'r') as f:
+            dev_examples = json.load(f)
+
+        word_vocab = pickle.load(
+            open(os.path.join(self._data_root_path, self._word_vocab_file_name), 'rb'))
+
+        char_vocab = pickle.load(
+            open(os.path.join(self._data_root_path, self._char_vocab_file_name), 'rb'))
+
+        return train_examples, dev_examples, word_vocab, char_vocab
+
+    def _save_processed_data(self, train_examples, dev_examples, word_vocab, char_vocab):
+        """Save processed data to disk
+
+        Parameters
+        ----------
+        train_examples : List[Tuple]
+            Processed train examples. Each tuple consists of question_id, record_index,
+            context_tokens_indices, question_tokens_indices, context_chars_indices,
+            question_char_indices, start_token_index_of_the_answer, end_token_index_of_the_answer,
+            context, context_tokens_spans
+        dev_examples : List[Tuple]
+            Processed dev examples. Each tuple consists of question_id, record_index,
+            context_tokens_indices, question_tokens_indices, context_chars_indices,
+            question_char_indices, start_token_index_of_the_answer, end_token_index_of_the_answer,
+            context, context_tokens_spans
+        word_vocab : Vocab
+            Word-level vocabulary
+        char_vocab : Vocab
+            Char-level vocabulary
+        """
+        with open(os.path.join(self._data_root_path, self._processed_train_data_file_name),
+                  'w') as f:
+            json.dump(train_examples, f)
+
+        with open(os.path.join(self._data_root_path, self._processed_dev_data_file_name), 'w') as f:
+            json.dump(dev_examples, f)
+
+        pickle.dump(word_vocab,
+                    open(os.path.join(self._data_root_path, self._word_vocab_file_name), 'wb'))
```

Comment:
for serialization, you can use [`vocab.to_json`](http://gluon-nlp.mxnet.io/master/api/modules/vocab.html?highlight=to_json#gluonnlp.Vocab.to_json)